framework/vulkan_sample.h

```diff
@@ -99,244 +101,281 @@ namespace vkb
  * - Core classes: Classes in vkb::core wrap Vulkan objects for indexing and hashing.
  */
 
+template <bool CppBindings>

```

Comment:
This might have reduced the downstream changes a bit.
But, again, as all the changes have been done already, I would prefer to not revert that again.
Besides that, it keeps the notion of `HPPVulkanSample` being some derivative of `VulkanSample`, which isn't the case.
Maybe, instead one could introduce some `using VulkanSampleC = VulkanSample<false>` and `using VulkanSampleCpp = VulkanSample<true>`, or so?